lib/compat.php

```diff
@@ -77,6 +77,84 @@ function gutenberg_provide_render_callback_with_block_object( $pre_render, $bloc
 }
 add_filter( 'pre_render_block', 'gutenberg_provide_render_callback_with_block_object', 10, 2 );
 
+/**
+ * Registers the user meta property responsible for storing preferences for the
+ * data script persistence middleware. The meta is explicitly registered so that
+ * it is made accessible for read and write via the REST API.
+ */
+function gutenberg_register_data_persistence_user_meta() {
+	global $wpdb;
+	register_meta(
+		'user',
+		$wpdb->prefix . 'data_persistence',
+		array(
+			'type'         => 'object',
+			'single'       => true,
+			'show_in_rest' => array(
+				'name'   => 'data_persistence',
+				'type'   => 'object',
+				'schema' => array(
+					'type'                 => 'object',
+					'properties'           => array(),
+					'additionalProperties' => true,
+				),
+			),
+		)
+	);
+}
+add_action( 'init', 'gutenberg_register_data_persistence_user_meta' );
+
+/**
+ * Enqueues inline scripts used to provide a custom storage interface for the
+ * data script persistence middleware. The custom storage implementation stores
+ * persisted values using user meta, bootstrapped for initial read at load-time,
+ * and updated via the REST API on change.
+ */
+function gutenberg_user_settings_data_persistence_inline_script() {
+	global $wp_scripts, $wpdb;
+
```

Comment:
What if there is no user set here. Might be worth checking if `$user_id` is not empty, just in case.

Reply:
> What if there is no user set here. Might be worth checking if `$user_id` is not empty, just in case.

Yeah, I agree. Will add.

Reply:
> > What if there is no user set here. Might be worth checking if `$user_id` is not empty, just in case.
> 
> Yeah, I agree. Will add.

Added in 5e0c5ebc7.

Noting: There aren't (but probably should be) similar such checks in the [core enqueueing of this script](https://github.com/WordPress/WordPress/blob/45286c5bb3f6fe5005567903ec858d87077eae2c/wp-includes/script-loader.php#L362)). Maybe it's expected that the `0` value of a logged-out user would be handled gracefully.